Controllers/CHumidityController.cpp

```diff
@@ -0,0 +1,195 @@
+    }
+
+    if (p_command->getName()->compare("heater") == 0)
+    {
+        result = overrideHeater(p_command);
+        b_command_recognised = true;
+    }
+    if (b_command_recognised)
+    {
+        switch (result)
+        {
+            case ICommand::COMMAND_OK:
+                //                p_comchannel->send("OK.\n");
+                break;
+            case ICommand::ERROR_ARG_COUNT:
+                p_comchannel->send("Wrong number of arguments.\n");
+                break;
+            case ICommand::ERROR_OUT_OF_BOUNDS:
+                p_comchannel->send("Argument out of bounds.\n");
+                break;
+            case ICommand::ERROR_TYPE_MISMATCH:
+                p_comchannel->send("Argument type mismatch.\n");
+                break;
+            default:
+                p_comchannel->send("Non-specific error with the command.");
+        }
+    }
+    return b_command_recognised;
+}
+
+void CHumidityController::reset()
+{
+    m_target_humidity = DISABLE_TARGET;
+    m_power_override = DISABLE_OVERRIDE;
+    m_control_loop.reset();
+}
+
+void CHumidityController::sendStatus(IComChannel *p_comchannel)
+{
+    etl::string<MAX_STRING_SIZE> message;
+    char value[10];
+    // send target humidity
+    message.assign("Target:      ");
+    sprintf(value, "%2.1f", m_target_humidity);
+    message.append(value);
+    p_comchannel->send(message);
+    // Send actual humidity
+    message.assign("Humidity: ");
+    sprintf(value, "%2.1f, ", m_humidity_sensor->getHumidity());
+    message.append(value);
+    p_comchannel->send(message);
+    // Send heater power
+    message.assign("Power:       ");
+    float power;
+    if (m_power_override == DISABLE_OVERRIDE)
+    {
+        power = mp_hw->getHardPwmOutput(CHANNEL_NUMBER - 1);
+    }
+    else
+    {
+        power = m_power_override;
+    }
+    sprintf(value, "%4.1f, ", power);
+    message.append(value);
+    p_comchannel->send(message);
+}
+
+/**
+ * @brief Set humidity value
+ */
+ICommand::command_error_code_t CHumidityController::setHumidity(
+    ICommand *p_command)
+{
+    // Sanitise command arguments
+    if (p_command->getArgumentCount != 1)
+    {
+        return ICommand::ERROR_ARG_COUNT;
+    }
+    m_target_humidity = *p_command;
+
+    if (m_target_humidity > MAX_HUMIDITY)
+    {
+        m_target_humidity = MAX_HUMIDITY;
+    }
+    if ((m_target_humidity < MIN_HUMIDITY) &&
+        (m_target_humidity != DISABLE_TARGET))
+    {
+        m_target_humidity = MIN_HUMIDITY;
+    }
+
+    return ICommand::COMMAND_OK;
+}
+
+ICommand::command_error_code_t CHumidityController::overrideHumidifier(
+    ICommand *p_command)
+{
+    // Sanitise command arguments
+    if (p_command->getArgumentCount() != 1)
+    {
+        return ICommand::ERROR_ARG_COUNT;
+    }
+    float power = *p_command;
+
+    // Execute command
+    if (power > MAX_POWER)
+    {
+        power = MAX_POWER;
+    }
+    if ((power < MIN_POWER) && (power != DISABLE_OVERRIDE))
```

Comment:
This is complex logic and would be much easier if you used boolean switch to control override.